src/game/option/option_control.c

```diff
@@ -426,6 +486,94 @@ static void Option_ControlChangeLayout(void)
     Config_Write();
 }
 
+static void Option_ControlProgressBar(TEXTSTRING *txt, int32_t timer)
+{
+    int32_t width = Text_GetWidth(txt);
+    int32_t height = TEXT_HEIGHT;
+
+    int32_t x = txt->pos.x;
+    int32_t y = txt->pos.y - TEXT_HEIGHT;
+
+    if (txt->flags.centre_h) {
+        x += (Screen_GetResWidthDownscaled() - width) / 2;
+    } else if (txt->flags.right) {
+        x += Screen_GetResWidthDownscaled() - width;
+    }
+    if (txt->flags.centre_v) {
+        y += Screen_GetResHeightDownscaled() / 2;
+    } else if (txt->flags.bottom) {
+        y += Screen_GetResHeightDownscaled();
+    }
+
+    int32_t percent = (timer * 100) / (FRAMES_PER_SECOND * BUTTON_HOLD_TIME);
+    CLAMP(percent, 0, 100);
+    Text_AddProgressBar(
+        txt, width, height, x, y, percent, g_Config.ui.menu_style);
```

Comment:
Do I need to add a check like:
```
if (percent) {
    Text_AddProgressBar(
        txt, width, height, x, y, percent, g_Config.ui.menu_style);
} else 
    Text_RemoveProgressBar();
}
```
? I think the way it's done now, the progress bar is always made which is why in `text.c` there's a double check for: `if (textstring->flags.progress_bar && textstring->progress_bar.value) {`.

Reply:
No need for now, might be a subject to change in the future if we need more progress bars that need to be displayed at 0%.